common/src/main/java/io/netty5/util/internal/NotNullByDefault.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.util.internal;
+
+
+import javax.annotation.Nonnull;
+import javax.annotation.meta.TypeQualifierDefault;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicates that parameters, return values of methods or fields contained in an annotated package should be considered
+ * as annotated with {@link Nonnull} annotation if it is not annotated with another nullness annotation.
+ */
+@Documented
+@TypeQualifierDefault({ ElementType.PARAMETER, ElementType.FIELD, ElementType.METHOD, ElementType.TYPE_USE })
+@Retention(RetentionPolicy.CLASS)
+@Target(ElementType.PACKAGE)
+@Nonnull
```

Comment:
`@org.jetbrains.annotations.NotNull` couldn't be applied because of its implementation detail.

Reply:
We could open an issue with them. Then clean up later, if they add what you're missing.

Reply:
Thanks for your reply. I agree. IMHO, It seems that the issue is related with IntelliJ's [impelmentation](https://github.com/JetBrains/intellij-community/blob/161e4339b5a32701cfe45bdddb9c2c03cb9d58fe/java/java-impl/src/com/intellij/codeInsight/annoPackages/Jsr305Support.java#L23) that IntelliJ only supports the nullness annotations under `javax.annotation.*` with `@TypeQualifierDefault`.

May I open an issue to ask if they will support `org.jetbrains.annotation.*` nullness annotations along with `@TypeQualifierDefault` or will have any plan to provide `@NotNullByDefault` equivalent?

+ If above annotation is acceptable, could I move on to adding the `@Nullable` annotations within the package?

Reply:
> May I open an issue to ask if they will support org.jetbrains.annotation.* nullness annotations along with @TypeQualifierDefault or will have any plan to provide @NotNullByDefault equivalent?

Yeah, please go ahead.

Reply:
I've just opened it ([link](https://youtrack.jetbrains.com/issue/IDEA-303505/orgjetbrainsNotNull-does-not-work-with-javaxannotationmetaTypeQualifierDefault)).

Reply:
Removed `ElementType.TYPE_USE` from `@TypeQualifierDefault` s value.
because when it was used with `@TypeQualifierDefault` it is very bug-prone.
and it effects type within the method body as well.